recorder.go

```diff
@@ -80,6 +80,12 @@ type Recorder struct {
 	// If nil, http.DefaultTransport is used.
 	Transport http.RoundTripper
 
+	// An optional Select function may be specified to control which recorded
+	// Entry is selected to respond to a given request. If nil, the default
+	// selection is used that picks the first recorded response with a matching
+	// method and url.
+	Select func(entries []Entry, req *http.Request) (Entry, bool)
```

Comment:
is it bad go style to give this function a typedef? that would clarify the declaration of `OncePerCall` below.

Reply:
Hrm, typedef is a good idea.

I struggled with two options:
1. An interface with a single function. This makes it easier for someone to make an implementation that keeps state, but makes simple function implementations harder.
2. Just the func. Someone can still make a stateful implementation and return a bound Method reference like I did below. Upside it that it makes more sense for stateful implementations, downside is that simple function implementations need a type adapter. e.g.
```
type Selector interface {
  Select(entries []Entry, req *http.Request) (Entry, bool)
}
...
type SelectorFunc func(entries []Entry, req *http.Request) (Entry, bool)
func (f SelectorFunc) Select(entries []Entry, req *http.Request) (Entry, bool) {
  return f(entries, req)
}
```
like https://godoc.org/net/http#HandlerFunc

Hard to say what's better... but I think using a typedef is a clear improvment.